src/main/java/net/wurstclient/commands/TpCmd.java

```
@@ -45,7 +48,36 @@ public void call(String[] args) throws CmdException
 		if(disableFreecam.isChecked() && WURST.getHax().freecamHack.isEnabled())
 			WURST.getHax().freecamHack.setEnabled(false);
 		
-		MC.player.setPos(pos.getX() + 0.5, pos.getY(), pos.getZ() + 0.5);
+		LocalPlayer player = MC.player;
+		ClientPacketListener netHandler = player.connection;
+		
+		double startX = player.getX();
+		double startY = player.getY();
+		double startZ = player.getZ();
+		
+		double dX = pos.getX() - startX;
+		double dY = pos.getY() - startY;
+		double dZ = pos.getZ() - startZ;
+		double totalDistanceSq = dX * dX + dY * dY + dZ * dZ;
+		

```

There are some known issues with teleportation using the burst packet methods.
1. The burst tp method may fail when the distance is < 10 blocks. Based on my testing, it will fail silently and does not trigger any error log on the console. This is very strange. Therefore, if the distance is less than 10 blocks, we resort back to the good old setPos method that never fails.

2. Despite the client can successfully teleport > 10 blocks away using burst packet method, it still triggers "moved too quickly check" somehow. However, it does not teleport the player back to the old position for whatever reasons.

3. The maximum teleportation distance using this method also depends on the server type (spigot, paper). But nevertheless it is a great enhancement on a vanilla or fabric minecraft server.